src/mongo/Config.class.php

```diff
@@ -1762,50 +1773,49 @@ protected function getConnectionForDataSource($dataSource)
         {
             $retries = 1;
             $exception = null;
-            $connected = false;
 
             do {
                 try {
-                    $this->connections[$dataSource] = $this->getMongoClient($ds['connection'], $connectionOptions);
-                    // Check if we are connected
-                    $connections = $this->connections[$dataSource]->getConnections();
-                    if (empty($connections) === false) {
-                        $connected = true;
-                    }
-                } catch (\MongoConnectionException $e) {
-                    self::getLogger()->error("MongoConnectionException attempt ".$retries.". Retrying...:" . $e->getMessage());
+                    $connectionString = $ds['connection'] . '?' . http_build_query($connectionOptions);
```

Comment:
It's validated in the `loadConfig` function. `$connectionOptions` is defensive just above this code - it only adds entries if things exist. And the `$ds['connection']` string HAS to be present. If it doesn't a `ConfigException` exception would be thrown much earlier in the execution.